services/workflows-service/src/workflow/workflow.service.ts

```diff
@@ -724,12 +724,7 @@ export class WorkflowService {
     const context = {
       ...workflow?.context,
       documents: workflow?.context?.documents?.map(
-        ({
-          // @ts-ignore
-          // Validating the context should be done without the propertiesSchema
-          propertiesSchema: _propertiesSchema,
-          ...document
-        }: DefaultContextSchema['documents'][number]) => {
+        (document: DefaultContextSchema['documents'][number]) => {
```

Comment:
Removed instances which removed `propertiesSchema` in favour of doing it inside of the validation method.